src/backend/replication/logical/origin.c

```diff
@@ -660,6 +670,15 @@ CheckPointReplicationOrigin(void)
 				 errmsg("could not write to file \"%s\": %m",
 						tmppath)));
 	}
+	if (chkp_size != sizeof(magic)) /* has some valid origins */
+	{
+		memcpy(buf + chkp_size, &crc, sizeof crc);
+		chkp_size += sizeof(crc);
+
+		/* NEON specific: persist snapshot in storage using logical message */
+		LogLogicalMessage("neon-file:pg_logical/replorigin_checkpoint", buf, chkp_size, false);
```

Comment:
Changes in this file are actually redundant as repl origins serve the reverse purpose -- tracking progress of replaying logical stream, not serving it. But neither should harm (and would be necessary if we want neon to be logical subscriber).